R/windsorize.R

```diff
@@ -1,10 +1,29 @@
 #' Windsorize
 #'
 #' Do some windsorization.
+#' 
+#' @param x A numeric vector containing real numbers
+#' @param p A number between 0 and 1, representing the mass of probability. Values are winsorized outside \code{p}. Default value is 0.9. 
+#' @return Windsorized vector of \code{x}.
+#' @examples
+#' windsorize(x = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
+#' windsorize(x = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10), p = 0.8)
+#' windsorize(c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10), 0.8)
 #' @export
+
 windsorize <- function(x, p = .90) {
-  q <- quantile(x, p)
-  x[x >= q] <- q
+  
+  if( sum(is.na(x))){
+    stop("Input x contains value NA")
```

Comment:
Do we need to throw an error here? What could be the workaround for `NA` values?